packages/next/src/cli/next-dev.ts

```diff
@@ -88,19 +88,25 @@ process.on('SIGTERM', handleSessionStop)
 
 let unwatchConfigFiles: () => void
 
-function watchConfigFiles(dirToWatch: string) {
+function watchConfigFiles(
+  dirToWatch: string,
+  onChange?: (filename: string) => void
+) {
   if (unwatchConfigFiles) {
     unwatchConfigFiles()
   }
 
   const wp = new Watchpack()
   wp.watch({ files: CONFIG_FILES.map((file) => path.join(dirToWatch, file)) })
   wp.on('change', (filename) => {
-    console.log(
-      `\n> Found a change in ${path.basename(
-        filename
-      )}. Restart the server to see the changes in effect.`
-    )
+    if (onChange) onChange(filename)
```

Comment:
onChange?.(filename)

Reply:
There is `else` branch. I guess I could have done it with default argument value being a function that logs stuff.